src/index/hnsw.cpp

```diff
@@ -585,6 +599,7 @@ HNSW::feedback(const DatasetPtr& query,
         return 0;
     }
 
+    std::shared_lock lock(rw_mutex_);
```

Comment:
thread-safe issues: L616 this->feedback maybe update  'conjugate_graph_', we need upgrade to unique lock ?